packages/functions/src/donor/DonorSwapAppointmentHandler.ts

```diff
@@ -0,0 +1,58 @@
+import { FunctionsApi } from "@zm-blood-components/common";
+
+import {
+  bookAppointment,
+  validateBookAppointment,
+} from "../common/BookAppointmentHelper";
+import {
+  cancelAppointment,
+  validateCancelAppointment,
+} from "../common/CancelAppointmentHelper";
+
+export default async function (
+  request: FunctionsApi.SwapAppointmentRequest,
+  callerId: string
+): Promise<FunctionsApi.SwapAppointmentResponse> {
+  const bookAppointmentValidation = validateBookAppointment(
+    request.bookAppointmentIds,
+    callerId
+  );
+
+  const cancelAppointmentValidation = validateCancelAppointment(
+    request.cancelAppointmentId,
+    callerId
+  );
```

Comment:
Do both of these fetch the appointment?

The should not. They should get it as an argument

Reply:
See above comment

Reply:
The extracted validation is diluted that way,
isn't there another option?